main.cpp

```diff
@@ -282,16 +203,37 @@ struct FloatType
     FloatType& multiply( float rhs );
     FloatType& divide( float rhs );
     
-
+    float pow(const IntType&);
+    float pow(const FloatType&);
+    float pow(const DoubleType&);
+    float pow(float);
+    
+    float powInternal(float* val, float arg);
+    
+    
 private:
     float* value;
-    
 };
 
+
+
 struct DoubleType
 {
     DoubleType(double doubleIn_) : value(new double(doubleIn_)){}
     
+    DoubleType(const DoubleType& other)
+    {
+        value = new double();
+        *value = *other.value;
+    }
+    
+    DoubleType& operator= (const DoubleType& other)
+    {
+        value = new double();
+        *value = *other.value;
+        return *this;
+    }
```

Comment:
You do not need these Rule of 3 functions, and they are implemented incorrectly.  You have a leak!
- [ ] remove these